common/geth/handle_error.go

```diff
@@ -65,15 +66,15 @@ func (f *FailoverController) handleError(err error) (NextEndpoint, ImmediateActi
 		var rpcError rpc.Error
 		if errors.As(err, &rpcError) {
 			ec := rpcError.ErrorCode()
-			f.Logger.Info("[JSON RPC Response Error]", "Error Code", ec, "Error", rpcError)
-			// we always attribute JSON RPC error as sender's fault, i.e no connection rotation
-			return CurrentRPC, Return
+			f.Logger.Warn("[JSON RPC Response Error]", "numRPCFfault", f.numberRpcFault, "errorCode", ec, "err", rpcError)
+			// we always attribute JSON RPC error as receiver's fault, i.e new connection rotation
+			return NewRPC, Return
 		}
 
 		// If no http response or no rpc response is returned, it is a connection issue,
 		// since we can't accurately attribute the network issue to neither sender nor receiver
 		// side. Optimistically, switch rpc client
-		f.Logger.Info("[Default Response Error]", err)
+		f.Logger.Warn("[Default Response Error]", "numRPCFfault", f.numberRpcFault, "err", err)
```

Comment:
Is it logged anywhere when it switches to a different RPC endpoint? That'd be a useful information to see from logs

Reply:
yes, logged